### DISCUSSION TITLE
Pagination with wrong placement

### DISCUSSION THREAD

**Discussion by nathipg** [ant-design/ant-design#41617] [Q&A] [answered]
Hello!

I'm upgrading my app code from v4 to v5 and I notice a diference with Pagination component, although there is no space to render the menu in the bottom, it is adding bottomLeft class. 

Example: https://codepen.io/nathipg/pen/qBMeyvB?editors=001
Note: Click on the select with "30 / page" content to see the problem

In version 4, the class topLeft was added, how can I reach this behavior again?

I saw that using pagination inside table it is possible, because you can define the placement, but using only Pagination I didn't find that option.

**Comment by Hidetoshi20** [comment, marked as answer]
## Solved by a prop in the source code 

I also encountered this problem, the placement error can not be specified, look at the source code and see an API to solve this problem  Looking at the source code of  `antd`, you can see that `props` is inherited from `rc pagination`

```tsx
export interface PaginationProps extends RcPaginationProps {
```

Keep looking at `rc pagination`

```tsx
export interface PaginationProps extends Partial<PaginationData> {
...
}
export interface PaginationData {
  ...
  selectComponentClass: React.ComponentType; // Drop-down box component
}
```

`rc pagination` itself does not have a select component and requires prop passing

Let's look at the `antd` render function

```tsx
return wrapSSR(
    <RcPagination
      ...
      selectComponentClass={selectComponentClass || (isSmall ? MiniSelect : MiddleSelect)} // You can customize select or else use MiniSelect or MiddleSelect
    />,
  );
```

Looking ahead to `MiniSelect`, which is also the `select` component of `antd`

```tsx
import type { SelectProps } from '../select';
import Select from '../select';

type CompoundedComponent = React.FC<SelectProps> & {
  Option: typeof Select.Option;
};

const MiniSelect: CompoundedComponent = (props) => <Select {...props} showSearch size="small" />;
const MiddleSelect: CompoundedComponent = (props) => <Select {...props} showSearch size="middle" />;

MiniSelect.Option = Select.Option;
MiddleSelect.Option = Select.Option;

export { MiniSelect, MiddleSelect };
```

This will solve our problem

```tsx
import { Select, SelectProps } from 'antd';

type CompoundedComponent = React.FC<SelectProps> & {
  Option: typeof Select.Option;
};
// The select of the native pagination component cannot automatically change the position in this and is always in the upper right corner
const SelectTopRight: CompoundedComponent = (props) => <Select {...props} placement="topRight" />;
SelectTopRight.Option = Select.Option;

export { SelectTopRight };
```

```tsx
<Pagination
  ...
  selectComponentClass={SelectTopRight} // Replace the default drop-down box
  />
```
